### PR TITLE
collector: list CRs instead of watching them

### PR DESCRIPTION
**Pull request by kopiczko**
Fixes https://github.com/giantswarm/giantswarm/issues/4601.

We think dupes are result of extra events during watch phase. See https://github.com/giantswarm/azure-operator/pull/429#issuecomment-454062640 for details.

### CODE REVIEW

File: service/collector/deployment.go [hunk 1]
```diff
@@ -85,90 +84,90 @@ func NewDeployment(config DeploymentConfig) (*Deployment, error) {
...
-	if err != nil {
-		return microerror.Mask(err)
+	var crs []providerv1alpha1.AzureConfig
+	{
+		mark := ""
+		page := 0
+		for page == 0 || len(mark) > 0 {
+			opts := metav1.ListOptions{
+				Continue: mark,
+			}
```
[the remark this answers is not in the document]
Comment (kopiczko) [reply service/collector/deployment.go]: I expect to have everything in a single page. How many clusters we can have? Like 200 would unrealistic given the constrains we have with each provide. I think only `pods` are objects we'd have to worry.

[the remark this answers is not in the document]
Comment (kopiczko) [reply service/collector/deployment.go]: I added paging for sanity.

**Comment by kopiczko** [comment]
`k8s API is still down` all the time :azureflip:. Merging.

**Comment by xh3b4sd** [comment]
Well, what are we going to do about the CI failures? Let's please not ignore it again. I am pretty happy with AWS since you sorted that. Let's have a strategy for Azure, in case we have to have one.

**Comment by kopiczko** [comment]
We need priority for that. So escalate with product and talk with our (magic) PO. W/o priority there isn't much we can do because those things take time.

Having said that I think this is just Azure slowness and increasing timeouts could help. Lately @r7vme said increasing timeouts to 80 min made all tests green. I've heard VPN gateway optimization reduced the startup time to 10 min but it's hard to believe because I remember before we had any form of connection between CP and tenant it was taking >30min.

**Comment by xh3b4sd** [comment]
Then let's increase the timeouts for Azure for now because the APIs there tend to take a lot of time, regardless our improvements.

**Pull request merged**